javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/ReferenceCollector.java

```diff
@@ -48,6 +49,14 @@
  * plugin.
  */
 public class ReferenceCollector {
+
+  // ReferenceCollector's classloader has a parent including the Gradle classpath, such as buildSrc
```

Comment:
It turns out we were previously loading SPI helper resources from buildSrc before our own classpath. Adding shadow to buildSrc, for use in the convention plugin, broke log4j instrumentation since shadow (probably incorrectly, it seems to be the log4j default) bundles in a log4j service file.